### DISCUSSION TITLE
How to set light theme and how to disable button on click text effect?

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#1247] [Q&A] [answered]
1) How to set light theme as the default instead of dark theme? Currently, I'm calling `self.action_toggle_dark()` at the end of `compose(self)` method.

2) When a button is clicked, there's a background color applied to the text - 'apple' in the screenshot shown below (and any previous such effect applied to another button is automatically removed). Is there a way to disable such effects - either partially for certain buttons or all of the buttons?

![image](https://user-images.githubusercontent.com/17766317/203321591-4278aee8-c910-4dee-ba5c-1684d8f56136.png)

**Comment by davep** [comment, marked as answer]
> 1. How to set light theme as the default instead of dark theme? Currently, I'm calling `self.action_toggle_dark()` at the end of `compose(self)` method.

The `App` class [has a `dark` property](https://textual.textualize.io/guide/design/#dark-mode); you could set that to `False`, perhaps in an `on_mount` method of the `App`.

```python
class LightApp( App[ None ] ):
    """An app that's always light."""

    ....

    def on_mount( self ) -> None:
        """Set up the application on startup."""
        self.dark = False
```

> 2. When a button is clicked, there's a background color applied to the text - 'apple' in the screenshot shown below (and any previous such effect applied to another button is automatically removed). Is there a way to disable such effects - either partially for certain buttons or all of the buttons?
> 
> ![image](https://user-images.githubusercontent.com/17766317/203321591-4278aee8-c910-4dee-ba5c-1684d8f56136.png)

This is controlled by the `text-style` of a `Button:focus` from CSS. So if you don't like the current style you can do something like:

```scss
Button:focus {
    text-style: bold;
}
```

By default in the Textual code it is:

```scss
Button:focus {
    text-style: bold reverse;
}
```

**Comment by learnbyexample** [comment]
Thanks a lot! I have a couple of follow ups:

1) Is `def on_mount( self )` the recommended way to initialize app settings - like choosing theme, initializing `self` variables, etc? I've been using `def __init__(self): super().__init__()` followed by other code as required (did this for `tkinter` projects before).

2) Is https://textual.textualize.io/guide/CSS/#pseudo-classes the relevant documentation for `Button:focus`? (found by searching for `focus`). If already available, could you point me to some more examples that use such features?

**Reply by davep** [reply]
1. Personally I'd do most app configuration in `on_mount` as it means all of your widgets will be up and running. In the case of setting `dark`, I've just tested to be sure and you can't actually set it in `__init__` without getting an exception -- presumably because light/dark mode is all about affecting the style of your application and, at that point, there's nothing to style yet. I have code that uses `__init__` for some things and `on_mount` for others and which goes where tends to be a case of "it depends".
2. Yup, that's the right spot in the docs to discover pseudo-classes. You'll find examples of their use throughout the Textual codebase; perhaps a good place to start would be having a read of [the examples](https://github.com/Textualize/textual/tree/main/examples) -- you'll see them used for dictionary and five_by_five.

**Comment by learnbyexample** [comment]
Thanks!

1) I had first called the theme change in `__main__` just before `app.run()` but that stopped working with the latest version. Then I tried `__init__` which failed as you note and then `compose()` had worked. Which is why I thought I should ask here since I was just trying out things without understanding.

2) I'll check out those examples.
